src/PrestaShopBundle/Command/GenerateHooksDocumentationCommand.php

```diff
@@ -0,0 +1,234 @@
+                '/docs/hooks/'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $dynamicHookConst = [
+            GridDefinitionHookByServiceIdsProvider::GRID_DEFINITION_HOOK_SUFFIX,
+            GridDefinitionHookByServiceIdsProvider::GRID_QUERY_BUILDER_HOOK_SUFFIX,
+            GridDefinitionHookByServiceIdsProvider::GRID_DATA_HOOK_SUFFIX,
+            GridDefinitionHookByServiceIdsProvider::GRID_FILTER_FORM_SUFFIX,
+            GridDefinitionHookByServiceIdsProvider::GRID_PRESENTER_SUFFIX,
+        ];
+
+        $hooks = $this->hookExtractor->findHooks(_PS_ROOT_DIR_);
+        $formatedHooks = [];
+        $formatedHooksKeys = [];
+        foreach ($hooks as $hook) {
+            $formatedHooks[$hook['hook']] = $hook;
+            $formatedHooksKeys[] = $hook['hook'];
+        }
+
+        if (!file_exists($this->hookFile)) {
+            throw new Exception(sprintf('File %s has not been found', $this->hookFile));
+        }
+
+        $hookFileContent = simplexml_load_file($this->hookFile);
+
+        $xmlHooks = $this->parseHooks($hookFileContent->entities);
+
+        foreach ($xmlHooks as $key => $xmlHook) {
+            if (in_array($xmlHook['hook'], $formatedHooksKeys)) {
+                $xmlHooks[$key]['type'] = $formatedHooks[$xmlHook['hook']]['type'];
+                $xmlHooks[$key]['file'] = $formatedHooks[$xmlHook['hook']]['file'];
+                $xmlHooks[$key]['aliases'] = $formatedHooks[$xmlHook['hook']]['aliases'] ?? [];
+                $xmlHooks[$key]['used_parameters'] = $formatedHooks[$xmlHook['hook']]['used_parameters'] ?? [];
+                $xmlHooks[$key]['full_implementation'] = $formatedHooks[$xmlHook['hook']]['full_implementation'];
+                $xmlHooks[$key]['locations'] = $formatedHooks[$xmlHook['hook']]['locations'] ?? [];
+                $xmlHooks[$key]['dynamic'] = false;
+            } elseif (!empty(array_filter($dynamicHookConst, fn ($str) => stripos($xmlHook['hook'], $str) !== false))) {
+                $xmlHooks[$key]['type'] = $this->defineHookType($xmlHook['hook']);
+                $xmlHooks[$key]['file'] = '';
+                $xmlHooks[$key]['aliases'] = [];
+                $xmlHooks[$key]['used_parameters'] = [];
+                $xmlHooks[$key]['full_implementation'] = '';
+                $xmlHooks[$key]['locations'] = ['back office'];
+                $xmlHooks[$key]['dynamic'] = true;
+            } else {
+                $xmlHooks[$key]['type'] = $this->defineHookType($xmlHook['hook']);
+                $xmlHooks[$key]['file'] = '';
+                $xmlHooks[$key]['aliases'] = [];
+                $xmlHooks[$key]['used_parameters'] = [];
+                $xmlHooks[$key]['full_implementation'] = '';
+                $xmlHooks[$key]['locations'] = ['back office'];
+                $xmlHooks[$key]['dynamic'] = false;
+            }
+        }
+
+        $outputDir = $input->getOption('output-dir');
```

Comment:
You should check that the value is not empty and that the folder is valid with an `is_dir` and throw an exception when it's not